src/libraries/System.Private.CoreLib/src/System/Globalization/DateTimeFormat.cs

```diff
@@ -1112,6 +1112,235 @@ private static StringBuilder FormatStringBuilder(DateTime dateTime, ReadOnlySpan
             return FormatCustomized(dateTime, format, dtfi, offset, result: null);
         }
 
+        internal static bool IsValidCustomDateFormat(ReadOnlySpan<char> format, bool throwOnError)
+        {
+            int length = format.Length;
```

Comment:
Caching span.Length like this into a local generally ends up being a deoptimization that inhibits removal of bounds checks.

Reply:
`span.Length` will always be a local at the Jit level anyway (because promotion).

Reply:
will fix that.